jormungandr/src/client.rs

```diff
@@ -47,37 +62,44 @@ fn handle_get_block_tip(blockchain_tip: &Branch) -> Result<Header, Error> {
 
 const MAX_HEADERS: usize = 2000;
 
-fn find_latest_checkpoint(storage: &Storage, checkpoints: &[HeaderHash]) -> Option<HeaderHash> {
-    // Filter out the checkpoints that don't exist
-    // (or failed to be retrieved from the store for any other reason)
-    // and find the latest by chain length.
-    checkpoints
-        .iter()
-        .filter_map(|hash| match storage.get_with_info(hash.clone()).wait() {
-            Ok(Some((_, info))) => Some((info.depth, hash)),
-            Ok(None) => None,
-            Err(_) => None,
-        })
-        .max_by_key(|&(depth, _)| depth)
-        .map(|(_, hash)| *hash)
+fn find_latest_checkpoint(
+    checkpoints: &[HeaderHash],
+    storage: &Storage,
+    block0_hash: &HeaderHash,
+) -> Result<HeaderHash, Error> {
+    // Filter out the checkpoints that don't exist in the storage;
+    // among the checkpoints present, find the latest by chain length.
```

Comment:
I think we should assume they are already sorted from the client point of view as the best to worse choice. Right?